libraries/service_helpers.rb

```diff
@@ -0,0 +1,58 @@
+module CernerSplunk
+  # Mixin Helper methods for the Splunk Service resource
+  module ServiceHelpers
+    include ResourceHelpers
+    require 'win32/service' if RUBY_PLATFORM =~ /mswin|mingw|windows/
+
+    # Reports if the Windows service for the current package exists
+    # @return [Boolean]
+    def service_exists
+      ::Win32::Service.exists?(process_name)
+    end
+
+    def splunk_bin_path
+      Pathname.new(install_dir).join('bin')
+    end
+
+    # Returns the process id of the daemon is running
+    # @return [Integer]
+    def service_pid
+      install_state = node.run_state['splunk_ingredient']['installations'][install_dir]
+      if install_state
+        pid_file = Pathname.new(install_dir).join('var/run/splunk/splunkd.pid')
+        if pid_file.exist?
+          pid = pid_file.readlines.first.to_i || false
+          return pid
+        end
+      end
+      false
+    end
+
+    # Reports if the daemon for the current package is running
+    # @return [Boolean]
+    def service_running
+      service_pid != false
+    end
+
+    # Reports the limits for the current package's running process
+    # @return [String] limits output
+    def service_limits
+      pid = service_pid
+      pid ? Pathname.new("/proc/#{pid}/limits").read : false
+    end
+
+    # Inserts ulimit command into the startup script for the current package's service
+    def try_write_initd_ulimit(ulimit)
```

Comment:
That was my initial idea, but it also updates the existing ulimit and if the file doesn't exist it doesn't actually change anything so I had a hard time coming up with a name that actually made sense for it.